### DISCUSSION TITLE
Disable indexing progress reports

### DISCUSSION THREAD

**Discussion by miquecg** [erlang-ls/erlang_ls#1349] [Support Requests] [answered]
Hi,

I'm using coc LSP client plugin on vim with erlang_ls. Every time I open a source file or type in a buffer several progress bars pop up reporting indexing of the project files. This is a bit annoying and first time I see it happening. Other Language Servers I tried, mainly Gleam and Elixir ones, don't show this behaviour.

I've posted a [discussion on coc.nvim repo](https://github.com/neoclide/coc.nvim/discussions/3920) but didn't find a solution yet, so I'm asking here just in case there's a configuration setting for erlang_ls to disable this.

**Comment by robertoaloi** [comment]
Hi @miquecg! The progress bar should only appear once, when you open the project for the first time.
The fact that you see multiple instances of it could indicate a project misconfiguration. Could you please share the structure of your project and your `erlang_ls.config` file?

We (should, unless there's a bug) respect the [client capabilities](https://github.com/erlang-ls/erlang_ls/blob/1d8500819002fa1a2d1969619dbbc99ca91409de/apps/els_lsp/src/els_work_done_progress.erl#L63) when it comes to progress reports. Could we verify the arguments that your client is sending to the server during initialization? You could enable DEBUG logging and show us the init request from [the logs](https://erlang-ls.github.io/troubleshooting/#logging).

**Reply by miquecg** [reply]
Hi :wave: 

> .config/erlang_ls/erlang_ls.config
```yaml
diagnostics:
  disabled:
    - dialyzer
    - elvis
```

> .vim/coc-settings.json
```json
{
  "coc.preferences.formatOnSaveFiletypes": [
    "c",
    "elixir",
    "erlang",
    "gleam"
  ],
  "codeLens.enable": true,
  "elixirLS.dialyzerEnabled": false,
  "languageserver": {
    "erlang": {
      "command": "erlang_ls",
      "filetypes": ["erlang"]
    },
    "gleam": {
      "command": "gleam",
      "args": ["lsp"],
      "filetypes": ["gleam"],
      "rootPatterns": ["gleam.toml", ".git"]
    }
  },
  "notification.disabledProgressSources": ["erlang"]
}
```

Project structure I think it doesn't matter in this case because it happens with any project (I just checked with `erlcloud` for instance). I should take a look at the client request during initialization yet.

Btw, I'm managing different Erlang versions with `asdf` and on every project I have a config overriding default ErlangLS path:

> .vim/coc-settings.json
```json
{
  "coc.preferences.formatOnSaveFiletypes": [],
  "languageserver.erlang.command": "~/.erlang_ls/release/25.0.4/bin/erlang_ls"
}
```

**Reply by miquecg** [reply]
Sorry for the late reply @robertoaloi. I've gathered more debug info from language server initialization in this [gist](https://gist.github.com/miquecg/e65dbed6d31009c7b74475a7c3ee6e08), using Rebar3 as a project sample.

**Comment by robertoaloi** [comment]
Hi @miquecg , from the gist it looks like the client has the workdone-progress-reports enabled:

https://gist.github.com/miquecg/e65dbed6d31009c7b74475a7c3ee6e08#file-coccommand-workspace-showoutput-L436

Not sure how to disable it in VIM, though.

**Reply by miquecg** [reply]
Hi again,

I've managed to get rid of the annoying pop-ups setting this in the config: `"notification.disabledProgressSources": ["*"]`.

**Comment by miquecg** [comment, marked as answer]
Just put this line into your `coc-settings.json`:

> "notification.disabledProgressSources": ["*"]

It should be possible to just supress progress for certain language servers, but [I still need to figure out](https://github.com/neoclide/coc.nvim/discussions/3920#discussioncomment-3659521).
